ring/ring.go

```diff
@@ -324,18 +330,46 @@ func (r *Ring) loop(ctx context.Context) error {
 	r.updateRingMetrics()
 	r.mtx.Unlock()
 
+	// Debounce WatchKey updates, as they can be frequent enough to cause lock
+	// contention. The most recent update is the one we'll use when we
+	// periodically update the ring.
+
+	go func() {
+		t := time.NewTicker(r.cfg.UpdateInterval)
```

Comment:
We should probably validate that `r.cfg.UpdateInterval` is > 0 in `New` function, or this will panic. Or use some sane positive value instead.

Reply:
I think if we don't turn a 0 interval into a sane default, this will be an annoying breaking change for other repos (incl. Mimir) when this PR is merged. As they construct ring.Config{} in their own tests.

Reply:
Makes sense, that should also minimize changes required to make tests pass.